### ISSUE TITLE
MP3 fails with AssertionError: APEv2 Footer preamble

### ISSUE THREAD

**Issue by jasonauk** [Borewit/music-metadata#362]
**Bug description**
I've tried many MP3s and I am unable to get them to parse in 5.2.0. I get the following error:

```
AssertionError: APEv2 Footer preamble
    at Function.parseTagFooter
    at Function.parseTagHeader
    at async f.parse
    at async g.parseID3v2
    at async Function._parse
```

This issue appears to have been introduced in 4.8.0. If I remove the following `if` block or roll back to 4.7.0 it works as expected: https://github.com/Borewit/music-metadata/blob/c66f1e3f5270685662c1d4d24e36426dae5f47ba/lib/id3v1/ID3v1Parser.ts#L123-L126

I can only assume the apeOffset is being calculated incorrectly.

The files have been tagged with MP3Tag (https://www.mp3tag.de/en/)

I can also repro here with drag-and-drop: https://audio-tag-analyzer.netlify.com/

**Expected behavior**
The MP3 parses and I can extract artist and title as needed.

**Audio file demonstrating the problem**
File attached.
[Band Aid - Do They Know It's Christmas_ (1984 Original Version).zip](https://github.com/Borewit/music-metadata/files/3983844/Band.Aid.-.Do.They.Know.It.s.Christmas_.1984.Original.Version.zip)

**Comment by Borewit** [comment]
In principle music-metadata parses audio files in a streaming manner. That implies a stream is read from the beginning to the end. To allow appending APEv2 being found in files, which require to read in the reverse direction, it pre-scans the last bit of a file, to look for a APE-footer.

In this case an APE-footer is indeed found.

Based on the APEv2-footer, the offset of the APEv2-header can be calculated.
This offset is `this.options.apeOffset` has a value of `3542880` in case of your file.

At this point the file is streamed to the parser(s), `this.options.apeOffset` proceed parsing the APE-header at that offset.

However, there is no APE-footer in this file, just a bit of garbage. If I tweak the code as if the APE-footer would be at that calculated position, it does extract APEv2 tags.

I was pretty sure it was caused by corrupted APEv2 tag, but I think your files are appended with [APEv**1** headers](http://wiki.hydrogenaud.io/index.php?title=APEv1_specification), which unlike [APEv2](http://wiki.hydrogenaud.io/index.php?title=APEv2_specification) have no APE header, just the footer.

Please note that writing APE tags to an .mp3 file is an arguable combination. If you do it, probably better to use APEv2. But I recommend to use ID3v2.4 or ID3v2.3 header instead and get rid of of APEv1, APEv2 and ID3v1 tag headers.

I will investigate the possibility of reading APEv1 headers.

**Comment by Borewit** [comment]
APEv1 implemented via PR #364, part of release [v5.3.0](https://github.com/Borewit/music-metadata/releases/tag/v5.3.0)
